.github/workflows/schedule.py

```diff
@@ -75,7 +89,105 @@ def _load_issue_template(path: str) -> tuple[dict[str, str], str]:
         return front_matter, f.read()
 
 
+def main(templates: list[tuple[str, datetime]], dry_run: bool) -> None:
+    for template, create_date in templates:
+        cmd_lookup, cmd_create = gh_commands(template, create_date)
+        if dry_run:
```

Comment:
The lookup command isn't modifying anything should it could be included in a dry run.

Generally speaking, "dry runs" are implemented by using the flag in strategic places, to surgically disable code that mutates data or has other side effects. This conditional is too high up in the call hierarchy. That way you can maximize the amount of code that is actually run during a dry run.